pennylane_lightning/core/simulators/lightning_gpu/StateVectorCudaManaged.hpp

```diff
@@ -289,6 +289,32 @@ class StateVectorCudaManaged
         applyOperation(opName, wires, adjoint, params, matrix_cu);
     }
 
+    /**
+     * @brief Apply a single gate to the state-vector. Offloads to custatevec
+     * specific API calls if available. If unable, attempts to use prior cached
+     * gate values on the device. Lastly, accepts a host-provided matrix if
+     * otherwise, and caches on the device for later reuse.
+     *
+     * @param opName Name of gate to apply.
+     * @param wires Wires to apply gate to.
+     * @param adjoint Indicates whether to use adjoint of gate.
+     * @param params Optional parameter list for parametric gates.
+     * @param matrix Gate data (in row-major format).
+     * @param mat_size The size of the gate matrix
+     */
+    void applyOperation(const std::string &opName,
+                        const std::vector<std::size_t> &wires, bool adjoint,
+                        const std::vector<Precision> &params,
+                        const ComplexT *matrix, const size_t mat_size) {
+        std::vector<CFP_t> matrix_cu(mat_size);
+        std::transform(matrix, matrix + mat_size, matrix_cu.begin(),
+                       [](const std::complex<Precision> &x) {
+                           return cuUtil::complexToCu<std::complex<Precision>>(
+                               x);
+                       });
+        applyOperation(opName, wires, adjoint, params, matrix_cu);
+    }
```

Comment:
As we discussed, please update the applyOperation function which signature asks for a std::vector matrix to call this new one. This way we can reduce code and improve maintainability.

Reply:
Done in f23ebbcc